src/model.ts

```diff
@@ -78,32 +78,32 @@ export interface Model<TSchema extends BaseSchema, TOptions extends SchemaOption
     options?: Omit<FindOptions<TSchema>, 'limit' | 'projection' | 'skip' | 'sort'>
   ) => Promise<boolean>;
 
-  find: <TProjection extends Projection<TSchema> | undefined>(
+  find: <TProjection extends Projection<TProjection, TSchema> | undefined>(
```

Comment:
TIL that you can use the same generic identifier in both the left side and right side of the `extends`. How is that even possible? Isn't there some kind of recursion limit?

Reply:
Circular generic constraints are really interesting and make quite a bit of sense, but they only currently work when they're non-immediate. Definitely pushes the boundaries of what TypeScript is able to deliver.

[Here is a really good thread](https://github.com/microsoft/TypeScript/issues/51011) if you want to read a bit more about them.

And there is a recursion limit, but it's pretty sizable and we'd hit that when building the schema path options already if we were going to hit it, so there's no additional risk here.